app/controllers/api/v1/book_suggestions_controller.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class BookSuggestionsController < ApiController
+      skip_before_action :authenticate_request
+      def create
+        @suggestion = BookSuggestion.new(suggest_params)
+        @suggestion.user_id = current_user.id if current_user
```

Comment:
You can do @suggestion.user = current_user, if the user is nil. the suggestion user will be nil too. Rails does the magic for the association

Reply:
gotcha